core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -349,27 +349,34 @@ private void validateStackAuthentication(StackV4Request source) {
         return convertedSet;
     }
 
+    private boolean shouldCreateLoadBalancers(StackV4Request source) {
+        return (StackType.DATALAKE.equals(source.getType()) &&
+                entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId())) ||
+                (StackType.WORKLOAD.equals(source.getType()) &&
+                        entitlementService.datahubLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId()));
+    }
+
     private Set<LoadBalancer> createLoadBalancers(StackV4Request source, Stack stack) {
         Set<LoadBalancer> loadBalancers = new HashSet<>();
         Set<TargetGroup> targetGroups = new HashSet<>();
-        // TODO expand this to data hubs
-        if (StackType.DATALAKE.equals(source.getType()) &&
-            entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
-            LOGGER.info("Setting up load balancers for stack {}", source.getName());
-            Set<String> knoxGatewayGroupNames = loadBalancerConfigService.getKnoxGatewayGroups(stack);
-            Set<InstanceGroup> knoxGatewayGroups = stack.getInstanceGroups().stream()
+        if (!shouldCreateLoadBalancers(source)) {
+            return loadBalancers;
```

Comment:
What you suggest is an anti-pattern. Elsewhere we typically follow early-return pattern (https://medium.com/swlh/return-early-pattern-3d18a41bba8) which has the advantage of simpler indentation, and simpler mental model to follow because the function reads as: 
 - if not enabled return early
 - rest of the logic.

Reply:
then:
```
if (!shouldCreate) {
  return Sets.of()
} else {
  initialize vars
  logic
  return loadBalancers
}
```
it's more readable. I look at the code, and I see there are 2 output by indentation, without really reading any code looking if there is any return.

Reply:
Ended up rebasing on top of #9675, which did not end up requiring this part of the change. So this discussion is obsolete now.